classes/log/EmailLogDAO.inc.php

```diff
@@ -59,19 +59,22 @@ function _getByEventType($assocType, $assocId, $eventType, $userId = null, $rang
 		];
 		if ($userId) $params[] = $userId;
 
-		$result = $this->retrieveRange(
-			$sql = 'SELECT	e.*
-			FROM	email_log e' .
-			($userId ? ' LEFT JOIN email_log_users u ON e.log_id = u.email_log_id' : '') .
-			' WHERE	e.assoc_type = ? AND
+		$baseSql = '
+			FROM email_log e' .
```

Comment:
I did this update basically everywhere: extracted the `select`/`order by` and re-added them at the end.